### PR TITLE
Catalan Number Series

### PR DESCRIPTION
**Pull request by yukty2000**
…er Of Code

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
Yes, I have  read the Contributing Guidelines on Pull Requests.
### Description
Motivation-
This is another application of the catalan number series - solved using tabulation method.
(Write your answer here)
### Checklist

- [ Yes] I've read the contribution guidelines.
- [ Yes] I've checked the issue list before deciding what to submit.
- [ Yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#1436 Add code for number of binary trees to dp folder for Winter Of Code

### CODE REVIEW

File: C-Plus-Plus/dp/number_of_binary_trees.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#include <bits/stdc++.h>
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/number_of_binary_trees.cpp:1]: Have a beginner description about the Algorithms and the Code Implementation in a Multi-Line Comment.

Comment (HarshCasper) [inline C-Plus-Plus/dp/number_of_binary_trees.cpp:16]: Have a beginner description about the Algorithms and the Code Implementation in a Multi-Line Comment.

File: C-Plus-Plus/dp/number_of_binary_trees.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+using namespace std;
+typedef long long int lli;
+// Find the number of binary trees possible given the size of the preorder sequence-n
+// Dynamic Programming Approach
+// Complexity : Time- O(n^2) ; Space- O(n)
+int main()
+{
+    int n;
+    cin>>n;
+    lli *dp=(lli *)malloc((n+1)*sizeof(lli));
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/number_of_binary_trees.cpp]: Have all your primary logic in another function. The main() function should strictly serve as your Driver Code.

File: C-Plus-Plus/dp/number_of_binary_trees.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    dp[0]=1,dp[1]=1;
+    //Since number of trees that can be formed for n=0/1 is 1
+    for(int i=2;i<=n;i++)
+    {
+        dp[i]=0;
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/number_of_binary_trees.cpp]: Use proper Variable Names. Prefer using Snake Case or Camel Case convention.

File: C-Plus-Plus/dp/number_of_binary_trees.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        for(int j=0;j<i;j++)
+        {
+            dp[i]+=(dp[j]*dp[i-j-1]);
+        }
+    }
+    //Calculation is same to that of catalan number
+    //Print the dp array to check for answers for values<n
+    /*for(int i=0;i<=n;i++)
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/number_of_binary_trees.cpp]: Don't use Single and Multi-Line Comments inconsistenly

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by yukty2000** [comment]
@HarshCasper sir, I have made the changes.

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Comment by yukty2000** [comment]
@HarshCasper  Sir, please merge my pr.

**Comment by HarshCasper** [comment]
Few more changes: 

- Resolve the Merge Conflicts. 
- Format the Code properly. 
- Use meaningful Variable Names.

**Comment by yukty2000** [comment]
Made the changes @HarshCasper sir.

File: C-Plus-Plus/dp/number_of_binary_trees.cpp (hunk 1 above)
Comment (HarshCasper) [reply C-Plus-Plus/dp/number_of_binary_trees.cpp:16]: Can you please justify the comments here? Please left align them.

File: C-Plus-Plus/dp/number_of_binary_trees.cpp [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+lli PossibleBinaryTrees(int number)
+{
+    lli *table=(lli *)malloc((number+1)*sizeof(lli));
+    table[0]=1,table[1]=1;                                      //Since number of trees that can be formed for n=0/1 is 1
+    
+    for(int loop1=2;loop1<=number;loop1++)                      //Calculating possible number of binary trees for all i<=n and storing them for future use-Tabulation Method
+    {
+        table[loop1]=0;
+        for(int loop2=0;loop2<i;loop2++)                        //We loop over the possible splits from 0 to i-1
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/number_of_binary_trees.cpp]: Keep the Comments above the Line of Code. Not beside them.

File: C-Plus-Plus/dp/number_of_binary_trees.cpp [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+{
+    int number;                                                 //Enter length of preorder sequence
+    cin>>number;
+    cout<<PossibleBinaryTrees(number)<<endl;
+    return 0;
+}
+/* Sample input:-
+        n = 5;
+        Output = 42;
+        */
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/number_of_binary_trees.cpp]: Justify the Comments. Left align them.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by yukty2000** [comment]
I have left aligned the comments and made the changes @HarshCasper sir.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**